agent/internal/fluent_intg_test.go

```diff
@@ -0,0 +1,258 @@
+			DefaultLoggingConfig: &model.DefaultLoggingConfig{},
+		},
+	})
+	assert.NilError(t, err, "error starting fluentbit")
+	sys := actor.NewSystem("")
+	sys.MustActorOf(actor.Addr("fluent"), f)
+
+	// WHEN a container prints some predefined logs
+	trialID := math.MaxInt16 + rand.Int31n(math.MaxInt16)
+	expected, actual := makeLogTestCase(int(trialID), aConf.AgentID)
+	runContainerWithLogs(t, actual, int(trialID), f.port)
+
+	// THEN fluent should parse all fields as expected and ship them to the mock master.
+	i := 0
+	for {
+		select {
+		case l := <-logBuffer:
+			assertLogEquals(t, l, expected[i])
+			i++
+			if i == len(expected) {
+				return
+			}
+		case <-time.After(time.Minute):
+			assert.Equal(t, i, len(expected), "not enough logs received after one minute")
+		}
+	}
+}
+
+func TestFluentLoggingElastic(t *testing.T) {
+	// GIVEN an elastic instance to accept logs
+	elastic, err := testutils.ResolveElastic()
+	assert.NilError(t, err, "unable to connect to master")
+
+	// AND a successfully started fluent actor
+	aConf := defaultAgentConfig()
+	f, err := newFluentActor(aConf, aproto.MasterSetAgentOptions{
+		MasterInfo:     aproto.MasterInfo{},
+		LoggingOptions: testutils.DefaultElasticConfig(),
+	})
+	assert.NilError(t, err, "error starting fluentbit")
+	sys := actor.NewSystem("")
+	sys.MustActorOf(actor.Addr("fluent"), f)
+
+	// WHEN a container prints some predefined logs
+	trialID := math.MaxInt16 + rand.Int31n(math.MaxInt16)
+	expected, actual := makeLogTestCase(int(trialID), aConf.AgentID)
+	runContainerWithLogs(t, actual, int(trialID), f.port)
+
+	// This is really unfortunate, but we don't query for logs until they're more than 10 seconds old
+	// to _try_ to avoid the trickiness involved with elastic's consistency model.
+	time.Sleep(10 * time.Second)
+
+	assert.NilError(t, elastic.WaitForIngest(testutils.CurrentLogstashElasticIndex()))
+
+	// THEN fluent should parse all fields as expected and ship them to elastic.
+	logs, _, err := elastic.TrialLogs(int(trialID), 0, 4, nil, nil)
+	assert.NilError(t, err, "failed to retrieve trial logs")
+	assert.Equal(t, len(logs), len(expected), "not enough logs received after one minute")
+	for i, l := range logs {
+		assertLogEquals(t, *l, expected[i])
+	}
+}
+
+func makeLogTestCase(trialID int, agentID string) ([]model.TrialLog, string) {
+	expected := []model.TrialLog{
+		{
+			TrialID:     trialID,
+			AgentID:     &agentID,
+			ContainerID: stringToPointer("goodcontainer"),
+			RankID:      intToPointer(4),
+			Log:         stringToPointer("\n"),
+			StdType:     stringToPointer("stdout"),
+		},
+		{
+			TrialID:     trialID,
+			AgentID:     &agentID,
+			ContainerID: stringToPointer("goodcontainer"),
+			RankID:      intToPointer(1),
+			Level:       stringToPointer("INFO"),
+			Log:         stringToPointer("Workload completed: <RUN_STEP (100 Batches): (580,6289,4)> (duration 0:00:01.496132)\n"), // nolint:lll // Anything to make these lines shorter would look worse.
```

Comment:
non-blocking: There's a minor convention that machine-readable comments should not have the leading spaces:
```suggestion
			Log:         stringToPointer("Workload completed: <RUN_STEP (100 Batches): (580,6289,4)> (duration 0:00:01.496132)\n"), //nolint:lll // Anything to make these lines shorter would look worse.
```
I can't find an authoritative source on that convention, but here are some mentions of it:
- https://golangci-lint.run/usage/false-positives/#nolint
- https://github.com/golang/go/issues/37974